src/marker.rs

```diff
@@ -680,6 +696,55 @@ impl MarkerExpression {
             (MarkerValue::QuotedString(l_string), MarkerOperator::Equal, MarkerValue::Extra) => {
                 extras.contains(l_string)
             }
+            (
+                MarkerValue::MarkerEnvVersion(MarkerValueVersion::PythonVersion),
+                operator,
+                MarkerValue::QuotedString(r_string),
+            ) => {
+                // ignore all errors block
+                (|| {
+                    // The right hand side is allowed to contain a star, e.g. `python_version == '3.*'`
+                    let (r_version, r_star) = Version::from_str_star(r_string).ok()?;
+                    let operator = operator.to_pep440_operator()?;
+                    // operator and right hand side make the specifier
+                    let specifier = VersionSpecifier::new(operator, r_version, r_star).ok()?;
+
+                    // We're passing a lot of versions here, e.g. 3.8 .. 3.100, but since we're
+                    // passing them in order it is expected that we either match early or not at all
+                    let compatible = python_versions
+                        .iter()
+                        .any(|l_version| specifier.contains(l_version));
```

Comment:
Not relevant for this PR but I wonder if it would make sense to have a `VersionSpeciefierSet` that can use a more optimized representation for storing many versions improved lookup. Or should we add the logic to `Specifier`: `specifier.matches(iter)`? 

Edit: Just saw your other comment. It may still make sense to have a `VersionSpecifierSet` that for now stores all versions internally in a `Vec`. We could then change its internals later if necessary.

Reply:
Good point, added `VersionSpecifiers::contains` (even though it's unrelated to the main PR)